include/evmc/evmc.hpp

```diff
@@ -294,15 +294,22 @@ constexpr address from_hex<address>(const char* s) noexcept
           byte(s, 14), byte(s, 15), byte(s, 16), byte(s, 17), byte(s, 18), byte(s, 19)}}};
 }
 
+/// A helper to report compile-time literal parsing errors. By design non-constexpr.
 template <typename T>
-constexpr T from_literal(const char* s)
+T error(const char*)
+{
+    std::abort();
+}
+
+template <typename T>
+constexpr T from_literal(const char* s) noexcept
 {
     return (s[0] == '0' && s[1] == '\0') ?
                T{} :
-               !(s[0] == '0' && s[1] == 'x') ?
-               throw "literal must be in hexadecimal notation" :
-               (length(s + 2) != sizeof(T) * 2) ? throw "literal must match the result type size" :
-                                                  from_hex<T>(s + 2);
+               !(s[0] == '0' && s[1] == 'x') ? error<T>("literal must be in hexadecimal notation") :
```

Comment:
Can somehow this be done via "static_assert" ?

Reply:
It might be possible in C++11 but after long night hours I still don't have it fully right. It should be much simpler in C++14 where you can just use variables in `constexpr` functions.